### ISSUE TITLE
Add support for Dockerfile --build-arg

### ISSUE THREAD

**Issue by wwwehr** [serverless/serverless#9197]
### Use case description

As a developer, I want to include the usage of `build-arg` in the docker build so that I can manage variables from `serverless.yml`.

Right now I need to use the scripts plugin to work around the limitations of the docker build.

[docs](https://docs.docker.com/engine/reference/commandline/build/#set-build-time-variables---build-arg)

### Proposed solution

Instrument the docker build command so that we can pass in configuration from `serverless.yml`

**Comment by insidewhy** [comment]
@wwwehr How did you use the scripts plugin to add buildArgs support before this was merged? I'd really like to be able to use `--secret` during the docker build.

**Comment by wwwehr** [comment]
> @wwwehr How did you use the scripts plugin to add buildArgs support before this was merged? I'd really like to be able to use `--secret` during the docker build.

Sorry to say that code is long in the past with a prior client and I no longer have access to readily help. I certainly used [the docs](https://www.serverless.com/framework/docs/guides/plugins/creating-plugins) and had built a plugin that was very much tailored to our machine learning workflow. I used serverless to manage and architect the aws step functions with containers, sagemaker, and or eventbridge.

So back to this question, you asked about the plugin - I'm pretty sure I just injected my new args into the existing serverless context at the earliest stage. If memory serves, the docker stuff (at the time) didn't obey the timelines. This may have improved.

It might be easy enough to just add your new args in the same way I added the new ones this PR.
